indico/modules/categories/controllers/management.py

```diff
@@ -428,15 +428,14 @@ class RHDeleteEvents(RHManageCategorySelectedEventsBase):
     def _process(self):
         is_submitted = 'confirmed' in request.form
         if not is_submitted:
-            # find out which active bookings are linked to the events in question
-            num_bookings = (Reservation.query
-                            .join(ReservationLink)
-                            .join(Event, Event.id == ReservationLink.linked_event_id)
-                            .filter(Event.id.in_(e.id for e in self.events),
-                                    Reservation.is_pending | Reservation.is_accepted)
-                            .count())
+            # find out which active booking occurrences are linked to the events in question
+            num_booking_occurrences = (ReservationOccurrence.query
+                                       .join(ReservationOccurrenceLink)
+                                       .join(Event, Event.id == ReservationOccurrenceLink.linked_event_id)
```

Comment:
I think this doesn't cancel occurrences linked to a contribution/session because in that case the `linked_event_id` is None and only the `event_id` is used (that one is always used when linking to an event or something inside).

The cancellation logic itself is correct though, since it uses `all_room_reservation_occurrence_links` for each event which uses the correct FK to get verything.